codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -553,13 +554,24 @@ void removeChild(State child) {
                 // Move up any grand children to be directly dependent on this node.
                 Iterator<IntObjectMap.PrimitiveEntry<State>> itr = child.children.entries().iterator();
                 while (itr.hasNext()) {
-                    takeChild(itr, itr.next().value(), false, events);
+                    //distribute the weight of child to it's dependency base on the proportion
+                    State dependency = itr.next().value();
+                    dependency.weight = (short) ((double) dependency.weight / totalWeight * child.weight);
```

Comment:
This is the first time I am reading this code, so unsure whether this is actually meaningful but does changing weight has to also consider `totalQueuedWeights` of the parent?

Reply:
I don't think so. `totalQueuedWeights` just contains a stream's child weights.

Reply:
To be more specific, `totalQueuedWeights` only contains the total weights of children in a stream's `pseudoTimeQueue`, if the `activeCountForTree` of a child is 0, it's weight will not been counted into `totalQueuedWeights`.